### ISSUE TITLE
report generation fails

### ISSUE THREAD

**Issue by duckinator** [how-is/how_is#25]
Backtrace:

```
PS C:\Users\Marie\dev\webpages\duckinator\how-is-rubygems> how_is --config
C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:6: warning: already initialized constant C
C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is.rb:7: warning: previous definition of C was here
C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/report.rb:41:in `issue_or_pr_summary': undefined method `strftime' for nil:NilClass (NoMethodError)
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/report.rb:56:in `block in <class:Report>'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/report/html.rb:60:in `instance_exec'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/report/html.rb:60:in `export'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/report.rb:80:in `export'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:60:in `block in from_config'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:56:in `each'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:56:in `map'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:56:in `from_config'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/lib/how_is/cli.rb:36:in `from_config_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/how_is-8.0.0/exe/how_is:56:in `<top (required)>'
        from C:/Ruby22/bin/how_is:22:in `load'
        from C:/Ruby22/bin/how_is:22:in `<main>'
```

Looks like `oldest['date']` in [line 41 of report.rb](https://github.com/how-is/how_is/blob/82ab6199/lib/how_is/report.rb#L41) is nil. This value is set in [lines 41-42 of analyzer.rb](https://github.com/how-is/how_is/blob/82ab6199827bad64cd471c03939af00a7728d3c8/lib/how_is/analyzer.rb#L41-L42).

This may be an issue of using Symbols vs using Strings for keys. See: [lines 160-168 of analyzer.rb](https://github.com/how-is/how_is/blob/82ab6199827bad64cd471c03939af00a7728d3c8/lib/how_is/analyzer.rb#L160-L168) vs [line 41 of report.rb](https://github.com/how-is/how_is/blob/82ab6199827bad64cd471c03939af00a7728d3c8/lib/how_is/report.rb#L41).

Aside: Even if Symbol vs String keys is a red herring (e.g. if they're normalized somehow), that should be made consistent to avoid confusion.

**Comment by duckinator** [comment]
I changed `oldest['date']` to `oldest[:date]` and it got farther before raising an error, so I'm pretty sure that's what the issue here is.

ofc, it's apparently broken in other ways, so...

**Comment by jm3** [comment]
fwiw, i get the same error with ruby 2.3.1 (rvm) on OS X 10.11.6:

```bash
% ruby --version
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

% rvm list

rvm rubies

   rbx-2.5.8 [ x86_64 ]
   ruby-1.9.3-p194 [ x86_64 ]
   ruby-2.0.0-p576 [ x86_64 ]
   ruby-2.1.2 [ x86_64 ]
   ruby-2.1.5 [ x86_64 ]
   ruby-2.2.3 [ x86_64 ]
   ruby-2.3.0 [ x86_64 ]
   ruby-2.3.0-preview1 [ x86_64 ]
=* ruby-2.3.1 [ x86_64 ]

% gem install how_is
Fetching: multipart-post-2.0.0.gem (100%)
Successfully installed multipart-post-2.0.0
Fetching: faraday-0.9.2.gem (100%)
Successfully installed faraday-0.9.2
Fetching: jwt-1.5.4.gem (100%)
Successfully installed jwt-1.5.4
Fetching: multi_xml-0.5.5.gem (100%)
Successfully installed multi_xml-0.5.5
Fetching: oauth2-1.2.0.gem (100%)
Successfully installed oauth2-1.2.0
Fetching: descendants_tracker-0.0.4.gem (100%)
Successfully installed descendants_tracker-0.0.4
Fetching: github_api-0.13.1.gem (100%)
Successfully installed github_api-0.13.1
Fetching: ttfunk-1.4.0.gem (100%)
Successfully installed ttfunk-1.4.0
Fetching: pdf-core-0.6.1.gem (100%)
Successfully installed pdf-core-0.6.1
Fetching: prawn-2.1.0.gem (100%)
Successfully installed prawn-2.1.0
Fetching: debug_inspector-0.0.2.gem (100%)
Building native extensions.  This could take a while...
Successfully installed debug_inspector-0.0.2
Fetching: spinny-utilities-0.2.0.gem (100%)
Successfully installed spinny-utilities-0.2.0
Fetching: default-0.2.0.gem (100%)
Successfully installed default-0.2.0
Fetching: spinny-global_variables-0.0.2.gem (100%)
Successfully installed spinny-global_variables-0.0.2
Fetching: lovama-0.2.4.gem (100%)
Successfully installed lovama-0.2.4
Fetching: letty-0.0.2.gem (100%)
Successfully installed letty-0.0.2
Fetching: spinny-quick_lambda-0.0.1.gem (100%)
Successfully installed spinny-quick_lambda-0.0.1
Fetching: mayhaps-0.3.0.gem (100%)
Successfully installed mayhaps-0.3.0
Fetching: spinny-0.8.3.gem (100%)
Successfully installed spinny-0.8.3
Fetching: openssl-better_defaults-0.0.1.gem (100%)
Successfully installed openssl-better_defaults-0.0.1
Fetching: curl_cacert-1.0.0.gem (100%)
Successfully installed curl_cacert-1.0.0
Fetching: tessellator-fetcher-4.0.1.gem (100%)
Successfully installed tessellator-fetcher-4.0.1
Fetching: how_is-8.0.0.gem (100%)
Successfully installed how_is-8.0.0
23 gems installed

% how_is jm3/dotfiles
/Users/jm3/.rvm/gems/ruby-2.3.1/gems/how_is-8.0.0/lib/how_is/cli.rb:6: warning: already initialized constant C
/Users/jm3/.rvm/gems/ruby-2.3.1/gems/how_is-8.0.0/lib/how_is.rb:7: warning: previous definition of C was here
/Users/jm3/.rvm/gems/ruby-2.3.1/gems/contracts-0.13.0/lib/contracts.rb:48:in `block in <class:Contract>': Contract violation for argument 1 of 1: (ParamContractError)
        Expected: (KeywordArgs[{:repository=>String, :from_file=>Optional[(String or nil)], :fetcher=>Optional[Class], :analyzer=>Optional[Class], :github=>Optional[Any]}]),
        Actual: {:repository=>"jm3/dotfiles", :from_file=>nil, :use_config_file=>false, :config_file=>nil}
        Value guarded in: HowIs::generate_analysis
        With Contract: KeywordArgs => Any
        At: /Users/jm3/.rvm/gems/ruby-2.3.1/gems/how_is-8.0.0/lib/how_is.rb:42
	from /Users/jm3/.rvm/gems/ruby-2.3.1/gems/contracts-0.13.0/lib/contracts.rb:154:in `failure_callback'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/gems/contracts-0.13.0/lib/contracts/method_handler.rb:143:in `rescue in block in redefine_method'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/gems/contracts-0.13.0/lib/contracts/method_handler.rb:136:in `block in redefine_method'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/gems/how_is-8.0.0/lib/how_is.rb:17:in `generate_report_file'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/gems/how_is-8.0.0/exe/how_is:75:in `<top (required)>'
	from /usr/local/bin/how_is:22:in `load'
	from /usr/local/bin/how_is:22:in `<main>'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/jm3/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by duckinator** [comment]
That doesn't appear to be the _same_ error, but that's still useful to know. Thanks for letting me know about it!

**Comment by lcguida** [comment]
I can confirm the issue on Mac OS `10.11.2`, ruby `2.2.1` and how_is `8.0.0`

```bash
gem list                                                                                              [10:36:59]

*** LOCAL GEMS ***

addressable (2.4.0)
bigdecimal (1.2.6)
bundler-unload (1.0.2)
contracts (0.14.0)
curl_cacert (1.0.0)
debug_inspector (0.0.2)
default (0.2.0)
descendants_tracker (0.0.4)
executable-hooks (1.3.2)
faraday (0.9.2)
gem-wrappers (1.2.7)
github_api (0.13.1)
hashie (3.4.4)
how_is (8.0.0)
io-console (0.4.3)
json (1.8.1)
jwt (1.5.4)
letty (0.0.2)
lovama (0.2.4)
mayhaps (0.3.0)
mini_magick (4.5.1)
multi_json (1.12.1)
multi_xml (0.5.5)
multipart-post (2.0.0)
oauth2 (1.2.0)
openssl-better_defaults (0.0.1)
pdf-core (0.6.1)
prawn (2.1.0)
psych (2.0.8)
rack (1.6.4)
rake (10.4.2)
rdoc (4.2.0)
rubygems-bundler (1.4.4)
rvm (1.11.3.9)
spinny (0.8.3)
spinny-global_variables (0.0.2)
spinny-quick_lambda (0.0.1)
spinny-utilities (0.2.0)
tessellator-fetcher (4.0.1)
thread_safe (0.3.5)
ttfunk (1.4.0)
```

```bash
ruby --version                                                                                        [10:36:56]
ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]
```

```bash
how_is rails/rails --report-file report.pdf                                                           [10:36:47]
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is/cli.rb:6: warning: already initialized constant C
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is.rb:7: warning: previous definition of C was here
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/exe/how_is:53:in `<top (required)>': invalid option: --report-file (OptionParser::InvalidOption)
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/how_is:23:in `load'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/how_is:23:in `<main>'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/ruby_executable_hooks:15:in `eval'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by duckinator** [comment]
@lcguida that issue is actually because you used `--report-file` instead of `--report` (hence "invalid option"), but if you use `--report` then you will likely get the error mentioned above.

**Comment by lcguida** [comment]
Yep. Using report I get the same errror:

```bash
how_is rails/rails --report report.pdf                                                                                                                                                                                        [11:36:09]
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is/cli.rb:6: warning: already initialized constant C
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is.rb:7: warning: previous definition of C was here
/Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/contracts-0.14.0/lib/contracts.rb:48:in `block in <class:Contract>': Contract violation for argument 1 of 1: (ParamContractError)
        Expected: (KeywordArgs[{:repository=>String, :from_file=>Optional[(String or nil)], :fetcher=>Optional[Class], :analyzer=>Optional[Class], :github=>Optional[Any]}]),
        Actual: {:repository=>"rails/rails", :from_file=>nil, :use_config_file=>false, :config_file=>nil}
        Value guarded in: HowIs::generate_analysis
        With Contract: KeywordArgs => Any
        At: /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is.rb:42
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/contracts-0.14.0/lib/contracts.rb:154:in `call'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/contracts-0.14.0/lib/contracts.rb:154:in `failure_callback'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/contracts-0.14.0/lib/contracts/method_handler.rb:143:in `rescue in block in redefine_method'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/contracts-0.14.0/lib/contracts/method_handler.rb:136:in `block in redefine_method'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/lib/how_is.rb:17:in `generate_report_file'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/gems/how_is-8.0.0/exe/how_is:75:in `<top (required)>'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/how_is:23:in `load'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/how_is:23:in `<main>'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/ruby_executable_hooks:15:in `eval'
	from /Users/lguida/.rvm/gems/ruby-2.2.1@how_is/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by duckinator** [comment]
This appears to be fixed, but the integration test is failing (even though running it manually works). Making a release anyway, since that's still better than the last release.

**Comment by duckinator** [comment]
Fixed in v9.0.
